python/lsst/summit/utils/efdUtils.py

```diff
@@ -447,13 +450,32 @@ def clipDataToEvent(df, event):
         The dataframe to clip.
     event : `lsst.summit.utils.efdUtils.TmaEvent`
         The event to clip to.
+    prePadding : `float`, optional
+        The amount of time before the nominal start of the event to include, in
+        seconds.
+    postPadding : `float`, optional
+        The amount of extra time after the nominal end of the event to include,
+        in seconds.
+    logger : `logging.Logger`, optional
+        The logger to use. If not specified, a new one is created.
 
     Returns
     -------
     clipped : `pd.DataFrame`
         The clipped dataframe.
     """
-    mask = (df['private_efdStamp'] >= event.begin.value) & (df['private_efdStamp'] <= event.end.value)
+    begin = event.begin.value - prePadding
+    end = event.end.value + postPadding
+    if begin < df['private_efdStamp'].min():
+        if logger is None:
+            logger = logging.getLogger(__name__)
```

Comment:
It would reduce code duplication to move this check for a logger before the `if begin...` statement, but I suppose that might be a slight performance hit if that makes you instantiate a logger a bunch of times when it is not needed.

Reply:
Yeah, I had the same thought - it's ugly this way, but felt slightly less bad than always making the logger even if it's not needed, and also better than making a flag, collecting logs, and logging after. I guess always making the logger is probably pretty cheap...

Reply:
OK, wow, it's _really_ fast - I guess that makes it an easy choice - will fix 🙂

![image](https://github.com/lsst-sitcom/summit_utils/assets/18741538/7df0d66d-c153-47e5-86eb-bbc5ac9ed699)